EternalBAND.Api/Services/AuthenticationService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using EternalBAND.Api.Options;
+using EternalBAND.DomainObjects;
+using EternalBAND.DomainObjects.ApiContract;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+
+namespace EternalBAND.Api.Services
+{
+    public class AuthenticationService
+    {
+        private readonly JwtTokenOptions jwtTokenSettings;
+        private readonly UserManager<Users> userManager;
+
+        public AuthenticationService(IOptions<JwtTokenOptions> jwtTokenSettings, UserManager<Users> userManager)
+        {
+            this.jwtTokenSettings = jwtTokenSettings.Value;
+            this.userManager = userManager;
+        }
+
+        // TODO we do not to mark this method async because token generation is almost static job,
+        // we may think to refactor async call as sync here
+        public async Task<string> CreateTokenAsync(string userName)
+        {
+            var user = await userManager.FindByEmailAsync(userName);
+            var roles = await userManager.GetRolesAsync(user);
+            return CreateToken(user, roles);
+        }
+
+        private string CreateToken(Users? user, IEnumerable<string> roles)
+        {
+            var tokenHandler = new JwtSecurityTokenHandler();
+            var key = Encoding.ASCII.GetBytes(jwtTokenSettings.Secret);
```

Comment:
to-do leak reminder JwtSecurityTokenHandler()

Reply:
Normally in .net we have IDisposible class which is identifier that derived class will be disposed right away when the life cycle complete for this purpose we are using **using** keyword.
(https://dotnetcode.medium.com/understanding-idisposable-in-net-for-resource-management-638ec99a19f6)
Here JwtSecurityTokenHandler is not using un managed memory, this class is doing simple things like validating token and creating token, so we do not be worry about memory leak here

Reply:
Perfect, thanks for the garbage Collections Gods. But still I do not like it. Memory instance references will be destroy thats okay but we can also set just 1 instance and thats enough. Let me explain you with a draw when we are available.

Reply:
added as singleton